common/common-api/src/main/java/org/ow2/proactive/utils/Lambda.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.utils;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.locks.Lock;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+
+// because we dont like checked exceptions
+@SuppressWarnings("squid:S00112")
+public class Lambda {
```

Comment:
We should rename this class, and maybe separate it into two classes (the `lock` methods and the rest are different concerns). First class could be CriticalSectionUtils, or CriticalSectionHandler, or simply CriticalSection, for the two `lock` methods. Second class could be for example LambdaWrapper, or FluentLambda, or AugmentedLambda, make it clear that it provides useful wrappers of common lambda usages.

Reply:
I would leave for now like this, maybe we will use some library for this